### PR TITLE
Fix a pickling test with spurious xfails

### PR DESCRIPTION
**Pull request by eerovaher**
### Description

There's a pickling test in `coordinates` that is not only ridiculously overcomplicated, but also causes 9 or 12 (depending on whether or not `scipy` is installed) spurious test xfails even though pickling the `astropy` objects in question works just fine.

- [x] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

File: astropy/coordinates/tests/test_pickle.py [hunk 1]
```diff
@@ -27,47 +32,24 @@ def test_pickle_longitude_wrap_angle():
...
-    [lambda *args: np.identity(3), coord.ICRS, coord.ICRS],
-    [0, 0],
-    [0],
-    [0],
-    [np.identity(3), coord.ICRS, coord.ICRS],
-]
-
-_kwargs = [
-    {"unit": "radian"},
-    {"z": 0.23},
```
Comment (neutrinoceros) [inline astropy/coordinates/tests/test_pickle.py:54]: I don't think these two are properly ported in this refactor yet

**Review by neutrinoceros — CHANGES_REQUESTED** [review]
overall an obvious :+1:, I'm just scrutinizing to make sure we get all details right;

File: astropy/coordinates/tests/test_pickle.py (hunk 1 above)
Comment (eerovaher) [reply astropy/coordinates/tests/test_pickle.py:54]: This is supposed to be a pickling test and these objects have no memory of how they were initialized.

Comment (neutrinoceros) [reply astropy/coordinates/tests/test_pickle.py:54]: thanks for clarifying

**Review by neutrinoceros — APPROVED** [review]

**Pull request merged**